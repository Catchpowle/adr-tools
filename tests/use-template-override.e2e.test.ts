/* eslint-disable no-sync */
import { describe, it, expect, afterEach, beforeEach } from 'vitest';
import * as childProcess from 'child_process';
import * as path from 'path';
import * as fs from 'fs/promises';
import * as os from 'os';

describe('Overriding templates', () => {
  const adr = path.resolve(path.dirname(__filename), '../src/index.ts');
  const command = `npx ts-node ${adr}`;

  let adrDirectory: string;
  let workDir: string;

  beforeEach(async () => {
    workDir = await fs.mkdtemp(path.join(os.tmpdir(), 'adr-'));
    adrDirectory = path.join(workDir, 'doc/adr');
    childProcess.execSync(`${command} init ${adrDirectory}`, { cwd: workDir });
  });

  afterEach(() => {
    childProcess.execSync(`rm -rf ${workDir}`);
  });

  it('should use an override template if one exists', async () => {
    await fs.mkdir(path.join(adrDirectory, 'templates'), { recursive: true });
    await fs.writeFile(path.join(adrDirectory, 'templates', 'template.md'), '# This is an override template\nTITLE\nDATE\nNUMBER\nSTATUS');

    childProcess.execSync(`${command} new Example ADR`, { cwd: workDir });
    childProcess.execSync(`${command} new Another Example ADR`, { cwd: workDir });

    const expectedFile: string = path.join(adrDirectory, '0002-example-adr.md');
    const expectedFile2: string = path.join(adrDirectory, '0003-another-example-adr.md');

    const fileContents = await fs.readFile(expectedFile, 'utf8');
    const fileContents2 = await fs.readFile(expectedFile2, 'utf8');

    expect(fileContents.replace(/\d{4}-\d{2}-\d{2}/, 'DATE-STRING-HERE')).toMatchSnapshot();
    expect(fileContents2.replace(/\d{4}-\d{2}-\d{2}/, 'DATE-STRING-HERE')).toMatchSnapshot();
  });
});
