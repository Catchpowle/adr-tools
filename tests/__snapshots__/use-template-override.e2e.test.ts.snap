// Vitest Snapshot v1

exports[`Overriding templates > should use an override template if one exists 1`] = `
"# This is an override template
Example ADR
DATE-STRING-HERE
2
Accepted"
`;

exports[`Overriding templates > should use an override template if one exists 2`] = `
"# This is an override template
Another Example ADR
DATE-STRING-HERE
3
Accepted"
`;
